### PR TITLE
Add identity property in managedCluster definition and add an example

### PR DESCRIPTION
**Pull request by norshtein**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by norshtein** [comment]
Is the CI broken? I run `npx autorest --reset && npx autorest --validation --azure-validator --message-format=json specification/containerservice/resource-manager/readme.md` locally and no error is reported.

**Comment by norshtein** [comment]
Hi @xizhamsft , could you please help have a review?

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-04-01/managedClusters.json [hunk 1]
```diff
@@ -1787,6 +1791,33 @@
...
+          "description": "The principal id of the system assigned identity which is used by master components."
+        },
+        "tenantId": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The tenant id of the system assigned identity which is used by master components."
+        },
+        "type": {
+          "type": "string",
+          "description": "The type of identity used for the managed cluster. Type 'SystemAssigned' will use an implicityly created identity in master components and an auto-created user assigned identity in MC_ resource group in agent nodes. Type 'None' will not use MSI for the managed cluster, service principal will be used instead.",
```
Comment (nschonni) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-04-01/managedClusters.json]:
```suggestion
          "description": "The type of identity used for the managed cluster. Type 'SystemAssigned' will use an implicitly created identity in master components and an auto-created user assigned identity in MC_ resource group in agent nodes. Type 'None' will not use MSI for the managed cluster, service principal will be used instead.",
```

**Comment by jhendrixMSFT** [comment]
Closing then re-open to fix CI.

**Comment by norshtein** [comment]
The CI keeps failed, and the error message seems to be unrelated to this PR. Anything I can do to fix this?  @jhendrixMSFT

**Comment by norshtein** [comment]
Close and reopen to fix CI.

**Comment by jhendrixMSFT** [comment]
@norshtein I think your branches are out-of-date.  Can you please rebase your PR on top of master and update the target branch as well?

**Comment by norshtein** [comment]
> @norshtein I think your branches are out-of-date. Can you please rebase your PR on top of master and update the target branch as well?

CC @xizhamsft . Could you please rebase branch `Azure:dev-containerservice-Microsoft.ContainerService-2019-04-01` upon master? I can then rebase this PR upon dev branch.

**Comment by xizhamsft** [comment]
/LGTM. 
@norshtein, could you try rebase your fork and update the PR? I don't own the dev branch. It will be the same process of rebased and then PR. Let me know if that does not work. Thanks.

**Comment by jhendrixMSFT** [comment]
Target branch has been rebased.

**Comment by norshtein** [comment]
Rebased upon updated target branch.

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**